x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Lookup.java

```diff
@@ -39,7 +38,7 @@ public class Lookup extends UnaryPlan {
     /**
      * References to the input fields to match against the {@link #localRelation}.
      */
-    private final List<NamedExpression> matchFields;
```

Comment:
These are always attributes (unresolved or resolved); the full generality of `NamedExpression` (including `Alias` etc.) makes reasoning about the match fields later down the road needlessly hard.